Fix demoman charging inconsistency (DRAFT)

allvei opened the pull request:
https://www.youtube.com/watch?v=m-Ls8trOy7Q

Creates a function which is used both on server and client to ensure proper sync, also fixing jitter and making turn speed consistent across different framerates, not tested on public.

wgetJane commented:
do you happen to have just the commits for fixing prediction for demoknight shield charging? i've attempted to fix it to submit as a pr for the source-sdk-2013 repo, but i didnt get too far and got too annoyed by it

i could get the initial charge itself to be predicted (so it doesnt rubberband after you right-click), however everything else i couldnt figure out yet (stopping a charge by running out of charge meter, stopping a charge by collision, and the charge meter regeneration itself is also unpredicted)

allvei commented:
@wgetJane I do, however, in testing, we found some inconsistencies with the server turn rate capping.

This might have been a version mismatch, as most of our testers ran fine. One of them hasn't been able to build the latest version successfully. Another one proved to be undetermined as to why it was not working properly.

allvei commented:
What I have done is:
- fix charging being framerate dependant
- updated the server method for limiting the charge turn rate
- removed tf_demoman_charge_frametime_scaling cvar as it won't be required any more
- removed m_filter
- added tf_charge_turn_rate FCVAR_REPLICATED
- added tf_charge_turn_tolerance FCVAR_REPLICATED
